mac_cleanup/core.py

```diff
@@ -208,28 +208,41 @@ def _count_dry(self) -> float:
         # Extracts paths from path_modules list
         path_list: list[Path_] = [path.get_path for path in path_modules]
 
-        # Set counter for estimated size
-        estimate_size: float = 0
-
         # Get thread executor
         executor = ThreadPoolExecutor()
 
         try:
             # Add tasks to executor
             tasks = [executor.submit(self._get_size, path) for path in path_list]
 
+            # Store paths and their corresponding futures
+            path_future_map = dict(zip(path_list, tasks))
```

Comment:
maybe just a list and next call instead of a getter?
```suggestion
            path_future_zip = list(zip(path_list, tasks))
```